server/base/src/main/java/org/apache/accumulo/server/conf/util/ZooPropSetTool.java

```diff
@@ -0,0 +1,391 @@
+  }
+
+  static class Opts extends ConfigOpts {
+
+    @Parameter(names = {"-d", "--delete"}, description = "delete a property")
+    public String deleteOpt = "";
+    @Parameter(names = {"-f", "--filter"},
+        description = "show only properties that contain this string in their name.")
+    public String filterOpt = "";
+    @Parameter(names = {"-fv", "--filter-with-values"},
+        description = "show only properties that contain this string in their name.")
+    public String filterWithValuesOpt = "";
+    @Parameter(names = {"--instanceName"},
+        description = "Specify the instance name to use. If instance name or id are not provided, determined from configuration (requires a running hdfs instance)")
+    public String instanceName = "";
+    @Parameter(names = {"--instanceId"},
+        description = "Specify the instance id to use. If instance name or id are not provided, determined from configuration (requires a running hdfs instance)")
+    public String instanceId = "";
+    @Parameter(names = {"-ns", "--namespace"},
+        description = "namespace to display/set/delete properties for")
+    public String namespaceOpt = "";
+    @Parameter(names = {"-nid", "--namespace-id"},
+        description = "namespace id to display/set/delete properties for")
+    public String namespaceIdOpt = "";
+    @Parameter(names = {"-s", "--set"}, description = "set a property")
+    public String setOpt = "";
+    @Parameter(names = {"-t", "--table"},
+        description = "table to display/set/delete properties for")
+    public String tableOpt = "";
+    @Parameter(names = {"-tid", "--table-id"},
+        description = "table id to display/set/delete properties for")
+    public String tableIdOpt = "";
+
+    private Filter filter = null;
+
+    @Override
+    public void parseArgs(String programName, String[] args, Object... others) {
+      super.parseArgs(programName, args, others);
+      var cmdMode = getCmdMode();
+      if (cmdMode == Opts.CmdMode.ERROR) {
+        throw new IllegalArgumentException("Cannot use set and delete in one command");
+      }
+      filter = new Filter(this);
+      if (!filter.getString().isEmpty()
+          && (cmdMode == Opts.CmdMode.SET || cmdMode == CmdMode.DELETE)) {
+        throw new IllegalArgumentException("Cannot use filter with set or delete");
+      }
+    }
+
+    Filter getFilter() {
+      return filter;
+    }
+
+    CmdMode getCmdMode() {
+      if (!deleteOpt.isEmpty() && !setOpt.isEmpty()) {
+        return CmdMode.ERROR;
+      }
+      if (!deleteOpt.isEmpty()) {
+        return CmdMode.DELETE;
+      }
+      if (!setOpt.isEmpty()) {
+        return CmdMode.SET;
+      }
+      return CmdMode.PRINT;
+    }
+
+    enum CmdMode {
+      ERROR, PRINT, SET, DELETE
+    }
+  }
+
+  /**
+   * The filter type and the filter string specified by command line options. Filter types are - no
+   * filter, filter on name or filter by name or value.
+   */
+  private static class Filter {
```

Comment:
I don't think you need any of this complexity for filtering the output of properties. You're reading all the properties anyway, so this isn't adding much value for the user. They can already pipe the output to grep. Can we try to minimize the feature and complexity of this tool with that fact in mind by getting rid of all the filtering stuff? It's overkill.

Reply:
It can be dropped - I was trying to mirror the same functionality that the shell provided.

Reply:
That makes sense, but I think we can target a subset of the functions... like for the ones we keep, we use the same syntax, but not add anything that they can already do themselves because of the fact that they're running it as a command-line tool instead of a shell command.

Reply:
Filtering removed in d11c27ea95